src/lib/buildPrompt.ts

```diff
@@ -25,12 +45,65 @@ export function buildPrompt(
 			)
 			.join("") + model.assistantMessageToken;
 
-	// Not super precise, but it's truncated in the model's backend anyway
-	return (
+	let webPrompt = "";
+
+	if (webSearch) {
+		const query = await getQueryFromPrompt(messages, model);
+		console.log("query: ", query);
+		const results = await searchWeb(query);
+
+		// fetch the top result, summarize it and pass it in the context
+		if (results.organic_results) {
+			const topUrl = results.organic_results[0].link;
+			console.log("top result :", topUrl);
+
+			// fetch the webpage
+			const htmlString = await fetch(topUrl)
+				.then((response) => response.text())
+				.catch((err) => console.log(err));
+
+			// extract the text content of the body element
+			const dom = new JSDOM(htmlString ?? "");
+			const body = dom.window.document.querySelector("body");
+
+			if (body === null) throw new Error("body is null");
+
+			const text = (naiveInnerText(body) ?? "").replace(/ {2}|\r\n|\n|\r/gm, "");
+
+			console.log(text);
+			// if there is a text, summarize it,
+			// TODO: This could be improved by using a dedicated summarization model if we have an endpoint for it.
```

Comment:
we could spawn one on Inference Endpoints if needed

Reply:
So far it seems to work okay like this. Maybe if we have a fine tuned model already available for summarization we could spawn one in the future but it's not a requirement to ship this, and I don't know if that's something we already have ?